src/EncumberableToken.sol

```diff
@@ -14,6 +14,11 @@ import { IERC7246 } from "./interfaces/IERC7246.sol";
  * @author Compound
  */
 contract EncumberableToken is ERC20, IERC20Permit, IERC7246 {
+    /**
+     * @dev Emitted when `taker` spends `amount` of their encumbrance from `owner`
+     */
+    event EncumbranceSpend(address indexed owner, address indexed taker, uint256 previousEncumbrance, uint256 newEncumbrance);
```

Comment:
I've tried to match the other events (e.g. `Release`, `Transfer`, `Approval`) by making this a noun. `EncumbranceSpend` is a lot more awkward a name than `Transfer`, however.

Additionally, I've opted to make this an event on the `EncumberableToken` contract, since the 7246 spec only includes the `Encumber` and `Release` events. Again, slightly awkward.

Open to suggestions on either count.

Reply:
To be fair, ERC-20 doesn't emit an event when an allowance is spent, so maybe we should reconsider whether an event is really needed here. Off-chain systems should be able to calculate encumbrance from just `Encumber`, `Release`, `Transfer`, and `Approval` events. If we do go ahead with an event, I think it should then be included in the ERC-7246 spec.

Reply:
Synced offline; agreed to delete this event.